src/Pipeline.cpp

```diff
@@ -658,8 +665,10 @@ void *Pipeline::compile_jit(const Target &target_arg) {
     // Compile to a module
     Module module = compile_to_module(args, name, target);
 
-    // Make sure we're not embedding any images
-    internal_assert(module.buffers().empty());
+    // We need to infer the arguments again, because compiling (GPU
+    // and offload targets) might have added new buffers we need to
+    // embed.
+    infer_arguments(module.functions().back().body);
```

Comment:
We've had complaints in the past about the overhead of calling realize... doesn't this make it much worse?